rust/arrow/src/array/array.rs

```diff
@@ -502,6 +502,98 @@ impl<T: ArrowNumericType> PrimitiveArray<T> {
     }
 }
 
+fn as_datetime<T: ArrowPrimitiveType>(v: i64) -> Option<NaiveDateTime> {
```

Comment:
This places these functions outside the `impl`, so that we can use them:

When used with `ArrowNumericType`, it can convert itself to i64 and thus this is simple.

When we use it with `PrimitiveType`, we use `to_usize` to convert to i64.